### PR TITLE
ACS-6356 Use larger runners on ES testing jobs

### PR DESCRIPTION
**Pull request by pzhyland**

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -303,8 +303,9 @@ jobs:
 
   tas_tests_search_api:
     name: ${{ matrix.testSuite }} | TAS tests (Search API)
-    runs-on: ubuntu-latest
+    runs-on: alfrescoPub-ubuntu2204-16G-4CPU
     if: >
+      (github.ref_name == 'master' || startsWith(github.ref_name, 'release/') || github.event_name == 'pull_request') &&
```
Comment (alxgomz) [inline .github/workflows/ci.yml:308]: I'm not sure this is useful given the `on.pull_request.branches` &  `on.push.branches` configs.

**Review by mpichura — APPROVED** [review]
LGTM (ES tests run times seem to have gone down)

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -366,7 +365,6 @@ jobs:
         with:
           persist-credentials: false
       - uses: Alfresco/alfresco-build-tools/.github/actions/get-build-info@v1.35.2
-      - uses: Alfresco/alfresco-build-tools/.github/actions/free-hosted-runner-disk-space@v1.35.2
```
Comment (gionn) [inline .github/workflows/ci.yml:369]: it is not worth anymore doing the cleanup? there is more space available with large runners?

Comment (pzhyland) [reply .github/workflows/ci.yml:369]: Yes, larger runners come with more disk space. What's more, the cleanup step on these runners might take couple of minutes:
https://github.com/Alfresco/acs-packaging/actions/runs/7084831853/job/19279893401#step:4

**Review by tpage-alfresco — APPROVED** [review]

**Pull request merged**